### PR TITLE
Implement the common config and validator

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

This PR seeks to implement the common config and validator.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-mongo/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	SCRAMSHA256AuthMechanism AuthMechanism = "SCRAM-SHA-256"
+	SCRAMSHA1AuthMechanism   AuthMechanism = "SCRAM-SHA-1"
+	MongoDBCRAuthMechanism   AuthMechanism = "MONGODB-CR"
+	MongoDBAWSAuthMechanism  AuthMechanism = "MONGODB-AWS"
+	X509AuthMechanism        AuthMechanism = "X.509"
+)
+
+// authMechanismMap holds a map with all possible auth mechanisms.
+// It's a helper for checking if a string is a valid auth mechanism.
+var authMechanismMap = map[string]AuthMechanism{
```
Comment (voscob) [inline config/config.go:40]: Just asking. 
Would it be better if you use the next map: `map[string]struct{}`?

And then just check it out like that:

```
if _, ok := authMechanismMap[strings.ToUpper(authMechanismStr)]; ok {
		return strings.ToUpper(authMechanismStr), nil
}
```

Comment (alyakimenko) [reply config/config.go:40]: Or in the case of `map[string]bool`, we could just do the following:
```go
if authMechanismMap[strings.ToUpper(authMechanismStr)] {
    return AuthMechanism(strings.ToUpper(authMechanismStr)), nil
}
```

Hmm, but actually we have a pretty small number of possible values for the `AuthMechanism`, so maybe we can just do:
```go
// IsValid checks if the underlying AuthMechanism is valid.
func (am AuthMechanism) IsValid() bool {
	switch am {
	case SCRAMSHA256AuthMechanism, SCRAMSHA1AuthMechanism, MongoDBCRAuthMechanism,
		MongoDBAWSAuthMechanism, X509AuthMechanism:
		return true
	}

	return false
}
```

And in the `Parse` function we'll check:
```go
// validate auth mechanism if it's not empty
if config.Auth.Mechanism != "" && !config.Auth.Mechanism.IsValid() {
	return Config{}, &InvalidAuthMechanismError{
		AuthMechanism: config.Auth.Mechanism,
	}
}
```

What do you think?

Comment (voscob) [reply config/config.go:40]: The last one is pretty damn good :)

And just a small question. Do we actually need the first check in the condition?
```config.Auth.Mechanism != ""```

File: config/config_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			authMechanismStr: "X.509",
+			want:             X509AuthMechanism,
+			wantErr:          false,
+		},
+		{
+			name:             "fail_unsupported",
+			authMechanismStr: "SASL",
+			want:             "",
+			wantErr:          true,
+		},
```
Comment (voscob) [inline config/config_test.go:72]: Should we check the empty string?

File: config/config.go (hunk 1 above)
Comment (alyakimenko) [reply config/config.go:40]: The `config.Auth.Mechanism` can be empty. In this case MongoDB will use the default authentication mechanism depending on a MongoDB server version.

File: config/errors.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import "fmt"
+
+// UnsupportedAuthMechanismError occurs when a string is not a valid [AuthMechanism].
+type UnsupportedAuthMechanismError struct {
```
Comment (voscob) [inline config/errors.go:20]: 👍

File: validator/validator_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "struct_with_key_tag",
+			args: args{
+				data: &struct {
+					Name string `key:"super_name"`
```
Comment (voscob) [inline validator/validator_test.go:35]: I just realized now why you used key tags in structures! That's a great idea. Kept it :)

File: config/config.go (hunk 1 above)
Comment (voscob) [reply config/config.go:40]: Oh yeah, right 👍

**Review by voscob — APPROVED** [review]
LGTM 👍

**Pull request merged**